components/brave_wallet_ui/page/screens/fund-wallet/deposit-funds.tsx

```diff
@@ -446,11 +448,17 @@ function DepositAccount() {
     selectedAsset ?? skipToken
   )
   const accountsForSelectedAssetCoinType = React.useMemo(() => {
-    return selectedAsset
-      ? accounts.filter((a) => a.accountId.coin === selectedAsset.coin)
-      : []
+    if (!selectedAssetNetwork) {
+      return []
+    }
+    return accounts.filter((a) =>
+      networkSupportsAccount(selectedAssetNetwork, a.accountId)
+    )
   }, [selectedAssetNetwork, accounts])
 
+  // mutations
+  const [generateReceiveAddress] = useGenerateReceiveAddressMutation()
```

Comment:
Consider changing this to a query

```suggestion
  const { data: receiveAddress } = useGeneratedReceiveAddressQuery(
    selectedAccount.accountId || skipToken,
    { refetchOnMountOrArgChange: true }
  )
```